docs/egress.md

```diff
@@ -0,0 +1,297 @@
+external network. The IPs in the pool can be allocated to the Egress resources
+as the Egress IPs. A typical ExternalIPPool resource example:
+
+```yaml
+- apiVersion: crd.antrea.io/v1alpha2
+  kind: ExternalIPPool
+  metadata:
+    name: prod-external-ip-pool
+  spec:
+    ipRanges:
+    - start: 10.10.0.2
+      end: 10.10.0.10
+    - cidr: 10.10.1.0/28
+    nodeSelector:
+      matchLabels:
+        network-role: egress-gateway
+```
+
+### IPRanges
+
+The `ipRanges` field contains a list of IP ranges representing the available IPs
+of this IP pool. Each IP range may consist of a `cidr` or a pair of `start` and
+`end` IPs inclusive.
+
+### NodeSelector
+
+The `nodeSelector` field specifies which Nodes the IPs in this pool can be
+assigned to. It's useful when you want to limit egress traffic to certain Nodes.
+The semantics of the selector is the same as those used elsewhere in Kubernetes,
+i.e. both `matchLabel` and `matchExpressions` are supported. It can be empty,
+which means all Nodes can be selected.
+
+## Usage examples
+
+### Configuring high available Egress
+
+In this example, we will make web apps in different namespaces use different
+egress IPs to access the external network.
+
+First, create an `ExternalIPPool` with a list of external routable IPs on the
+network.
+
+```yaml
+- apiVersion: crd.antrea.io/v1alpha2
+  kind: ExternalIPPool
+  metadata:
+    name: external-ip-pool
+  spec:
+    ipRanges:
+    - start: 10.10.0.11  # 10.10.0.11-10.10.0.20 can be used as Egress IPs
+      end: 10.10.0.20
+    nodeSelector: {}     # All Nodes can be Egress Nodes
+```
+
+Then create two `Egress` resources, each of which applies to web apps in one
+Namespace.
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: Egress
+metadata:
+  name: egress-prod-web
+spec:
+  appliedTo:
+    namespaceSelector:
+      matchLabels:
+        kubernetes.io/metadata.name: prod
+    podSelector:
+      matchLabels:
+        app: web
+  externalIPPool: external-ip-pool
+---
+apiVersion: crd.antrea.io/v1alpha2
+kind: Egress
+metadata:
+  name: egress-staging-web
+spec:
+  appliedTo:
+    namespaceSelector:
+      matchLabels:
+        kubernetes.io/metadata.name: staging
+    podSelector:
+      matchLabels:
+        app: web
+  externalIPPool: external-ip-pool
+```
+
+List the `Egress` resource with kubectl. The output shows each Egress gets one
+IP from the IP pool and gets one Node assigned as its Egress Node.
+
+```yaml
+# kubectl get egress
+NAME                 EGRESSIP       AGE   NODE
+egress-prod-web      10.10.0.11     1m    node-4
+egress-staging-web   10.10.0.12     1m    node-6
+```
+
+Now, the packets from the Pods with label `app=web` in `prod` Namespace to the
+external network will be redirected to `node-4` Node and SNATed to `10.10.0.11`
+while the packets from the Pods with label `app=web` in `staging` Namespace to
+the external network will be redirected to `node-6` Node and SNATed to
+`10.10.0.12`.
```

Comment:
in the `prod` Namespaces
to the `node-4` Node
in the `staging` Namespaces
to the `node-6` Node

Reply:
done. I assume you mean "in the `prod` Namespace", not "Namespace**s**"? I used namespace name label so it can only match one namespace.